app/src/main/java/com/example/pwmobilesdk/MainActivity.kt

```diff
@@ -192,6 +192,8 @@ class MainActivity : AppCompatActivity() {
                     }
 
                     _textViewLog?.append(lineToLog + "\n")
```

Comment:
I'd suggest to add one new method `logToConsole` that appends the TextViewLog and invokes the `scrollTextViewLog` at the end of it's method body. This way, `scrollTextViewLog ` would only need to be invoked in one central place. :)

Reply:
Update: I'd recommend to extract one new method `logToConsole` anyway -- even if the `scrollTextViewLog` will be removed as the scroll-movement can be set as a general behaviour of the ScrollView in the `MainActivity.onCreate` method :)